src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx

```diff
@@ -246,54 +324,58 @@ export const K8SResourceList = ({
         )
     }
 
-    const emptyPageActionHandler = () => {
-        clearSearch()
-        clearFilters()
+    const emptyStateActionHandler = () => {
+        setFilteredResourceList(resourceList?.data)
+        setSearchText('')
+        const pathname = `${URLS.RESOURCE_BROWSER}/${clusterId}/${ALL_NAMESPACE_OPTION.value}/${selectedResource.gvk.Kind.toLowerCase()}/${group}`
+        updateK8sResourceTab(pathname)
+        setSelectedNamespace(ALL_NAMESPACE_OPTION)
     }
 
     const renderEmptyPage = (): JSX.Element => {
-        if (noResults) {
-            return (
-                <ResourceListEmptyState
-                    title={RESOURCE_EMPTY_PAGE_STATE.title(selectedResource?.gvk?.Kind)}
-                    subTitle={RESOURCE_EMPTY_PAGE_STATE.subTitle(
-                        selectedResource?.gvk?.Kind,
-                        selectedResource?.namespaced,
-                    )}
-                />
-            )
-        }
-        return (
+        const isFilterApplied = searchText || location.search || selectedNamespace.value !== ALL_NAMESPACE_OPTION.value
+        return isFilterApplied ? (
             <ResourceListEmptyState
                 title={RESOURCE_LIST_EMPTY_STATE.title}
                 subTitle={RESOURCE_LIST_EMPTY_STATE.subTitle(selectedResource?.gvk?.Kind)}
-                actionHandler={emptyPageActionHandler}
+                actionHandler={emptyStateActionHandler}
+            />
+        ) : (
+            <ResourceListEmptyState
+                title={RESOURCE_EMPTY_PAGE_STATE.title(selectedResource?.gvk?.Kind)}
+                subTitle={RESOURCE_EMPTY_PAGE_STATE.subTitle(selectedResource?.gvk?.Kind, selectedResource?.namespaced)}
             />
         )
     }
 
     const changePage = (pageNo: number) => {
         setResourceListOffset(pageSize * (pageNo - 1))
 
-        // scroll to top on page change
-        if (resourceListRef.current) {
-            resourceListRef.current.scrollTo({ top: 0, behavior: 'smooth' })
-        }
+        /* NOTE: scroll to top on page change */
+        resourceListRef.current?.scrollTo({ top: 0, behavior: 'smooth' })
     }
 
     const changePageSize = (size: number) => {
         setPageSize(size)
         setResourceListOffset(0)
     }
 
+    if (resourceListDataError && !getIsRequestAborted(resourceListDataError)) {
```

Comment:
We should not write it like this

Reply:
Should we not write the showError in the render body (Eshank had told me he writes the showError in the service layer itself) or is something wrong with the if condition?

Reply:
Moved the showError into the function passed to useAsync using a try catch block